src/ol/renderer/canvas/canvasmaprenderer.js

```diff
@@ -193,11 +227,15 @@ ol.renderer.canvas.Map.prototype.renderFrame = function(frameState) {
       continue;
     }
     if (layerRenderer.prepareFrame(frameState, layerState)) {
+      if (layer instanceof ol.layer.Vector && layer.getSource().getWrapX()) {
```

Comment:
Maybe we can revisit this in the future... For now I'd just add a comment in the code to explain the strategy chosen here (repeat the feature overlays as soon as there's a vector layer with wrapX set to true).